### DISCUSSION TITLE
Replacing corner button with my own icon and action

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2638] [Q&A]
I want to replace the corner button with my own icon and with my own action (rather then opening backstage menu).  I was able to replace other toolbars by disabling default and defining my own using provideToolbarButtonItems().  But I am unable to find how I can replace the cornerItem button.

![CornerButton](https://user-images.githubusercontent.com/83007711/140039052-5d29cd6b-d1d1-43ba-adfe-01d15321b51d.JPG)

I found ToolWidgetComposer component which has ability to specify your own cornerItem and other toolbars.  This seems to be doing same thing as provideToolbarButtonItems() which I am using to define my toolbars (instead of ToolWidgetComposer).

Can you please guide how I can replace the cornerItem with my own button/toolbar.

Thank you.

**Comment by bsteinbk** [comment]
@ovaisa If this is an application that you created using CRA and the Bentley template then you only have the ability to hide the corner item or use the default corner item. If you did not use the CRA template please provide me details on how you specified your Frontstage and I can provide guidance on how the corner button can be changed.   @aruniverse we should address this shortcoming in the viewer for our 3.0 release.

**Reply by ovaisa** [reply]
Yes, I created this app using the CRA template (2.19.20)

**Reply by bsteinbk** [reply]
@ovaisa Sorry for inconvenience, I will work with @aruniverse  to get this addressed for 3.0.

**Reply by ovaisa** [reply]
Thank you @bsteinbk.  I think I should be able to achieve what I want by doing it in the backstage menu.  I would appreciate if you can point me to sample code that shows how to customize the backstage menu.

**Reply by bsteinbk** [reply]
@ovaisa - Your UiItemsProvider that provides tool buttons can also provide backstage items.  See [example](https://github.com/iTwin/itwinjs-core/blob/112933a965e339f36b8af4f092fae8c73f6767da/test-apps/ui-test-extension/src/ui/ExtensionUiItemsProvider.tsx#L37) of adding a backstage item that opens up a specific stage. You can also look at this [example](https://github.com/iTwin/itwinjs-core/blob/007be59f93813bc5525b1fe2afc48f20761bca56/test-apps/ui-test-app/src/frontend/appui/backstage/AppBackstageComposer.tsx#L59) of creating a backstage item that runs an action.  The actions typically activates a modal stage where you can presents any content you want.  The screen shot below shows a modal stage for displaying data and is opened via this [line](https://github.com/iTwin/itwinjs-core/blob/007be59f93813bc5525b1fe2afc48f20761bca56/test-apps/ui-test-app/src/frontend/appui/backstage/AppBackstageComposer.tsx#L61).
![backstage2](https://user-images.githubusercontent.com/65047615/140067011-6d879d9e-c20d-4856-b966-b91e6561f0a3.png)

**Reply by ovaisa** [reply]
I am able to add my items to the backstage, thank you for the links.  Is it possible to remove the two default or at least the user item in the default backstage.

**Reply by bsteinbk** [reply]
The user info item will be gone in 3.0.  A hack before 3.0, that may work, is to add the following to your .scss file
```
.nz-backstage-userProfile {
display: none !important;
}
```

**Reply by ovaisa** [reply]
Thank you Bill, this works...  By any chance is there a way to suppress the Frontstage tile - if not no worries, I can live with this.  Though if you can let me know the 'priority' of this tile, I can put mine at higher priority (so they show up first).

Can you also please advise the recommended way to put an HTML item on the screen.  I would like to display our product logo (HTMLImageElement) at bottom right of the Viewport.

**Reply by kckst8** [reply]
You are able to replace the corner item with a custom React component via the defaultUiConfig.contentManipulationTools.cornerItem.item prop:

![image](https://user-images.githubusercontent.com/6283674/140320398-3ed84b65-f19e-450c-9bfc-3d0efa04d6c5.png)

**Reply by bsteinbk** [reply]
The itemPriority for the "default" backstage item is 10 and its groupPriority is 100.  You can use negative values for both to ensure your entries or displayed first.
